zapcore/encoder.go

```diff
@@ -329,6 +329,9 @@ type EncoderConfig struct {
 	// Unlike the other primitive type encoders, EncodeName is optional. The
 	// zero value falls back to FullNameEncoder.
 	EncodeName NameEncoder `json:"nameEncoder" yaml:"nameEncoder"`
+	// Configures the field separator used by the console encoder. Defaults
+	// to tab.
+	ConsoleSeparator byte `json:"consoleSeparator" yaml:"consoleSeparator"`
```

Comment:
Should we use a string type for the separator so it's more flexible in the future? A `byte` is efficient, but will have issues if there's some non-ASCII character that someone would like to use.

cc @abhinav what do you think?

Reply:
Agree, As of today, I think the string makes more sense for the future.  @prashantv @abhinav  let me know if I should go ahead to change to the string.

Reply:
Yeah, a string separator makes sense.

Reply:
@abhinav @prashantv  done please have another look